nixos/modules/services/networking/radicale.nix

```diff
@@ -43,7 +54,7 @@ in
       '';
     };
 
-    services.radicale.config = mkOption {
+    config = mkOption {
```

Comment:
`settings` should also work with older versions, but I kept `config` for backwards compatibility

Reply:
The problem with this is that third-party modules can't rely on `settings`. Every module that wants to read or write `settings` options has the chance to not work in case the user doesn't use `settings`. And it's even worse: A third-party module writing to `settings` makes the `settings == {}` check fail, meaning the users `config` option gets ignored! Because of this I think it might be better to actually remove this option.

Reply:
> And it's even worse: A third-party module writing to `settings` makes the `settings == {}` check fail, meaning the users `config` option gets ignored! Because of this I think it might be better to actually remove this option.

That's not true: If both `config` and `settings` are set an assertion is triggerred.
Nevertheless, I'm all in favor of removing `config` but it might make sense to first mark it as deprecated using `lib.warn`.

Reply:
Agreed